internal/otelcollector/config/metric/agent/prometheus_receiver.go

```diff
@@ -140,7 +141,12 @@ func makeTLSConfig(istioCertPath string) *TLSConfig {
 	}
 }
 
-func makePrometheusIstioConfig() *PrometheusReceiver {
+func makePrometheusIstioConfig(envoyMetricsEnabled bool) *PrometheusReceiver {
+	metricNames := "istio_.*"
+	if envoyMetricsEnabled {
+		metricNames = strings.Join([]string{"envoy_.*", metricNames}, "|")
```

Comment:
Wouldn't a simple concatenation be easier?
```suggestion
		metricNames = "envoy_.*|" + metricNames
```

Reply:
Sure, this is simple and easy, but it is not an efficient way to concatenate strings.